### PR TITLE
Kyle Meltzer Testing

### PR DESCRIPTION
**Pull request by meltzerkyle**
### Assignment

- [ ] Use `yarn` to install the necessary dependencies from the `package.json` file.
- [ ]  type `yarn test` to start the test runner.
- [ ] open the `basic-javascript` folder and use `project-1.spec.js` to write tests for the functions inside `project-1.js`.
- [ ]  use `project-2.spec.js` to write tests for the functions inside `project-2.js`.

As you write your tests, bugs in the code may be revealed, feel free to correct and _refactor_ the source code to sitisfy your tests.

## Stretch Goal

- [ ]  Open the `advanced-javascript` folder and write tests for the `arrays.js` and `object.js` files inside the corresponding `*.spec.js` files.

### CODE REVIEW

**Comment by kelfro** [comment]
I know you were a little behind schedule yesterday, so try to finish at least 1 or 2 basic tests for each function in basic js today if you can. If you have enough time, you can work more on your edge cases (which look great so far!). Nice work!

File: basic-javascript/project-1.spec.js [hunk 1]
```diff
@@ -1,3 +1,332 @@
...
+});
+
+it("should convert strings to numbers if possible", () => {
+  expect(helpers.subtract("91", 1)).toBe(90);
+  expect(helpers.subtract(91, "1")).toBe(90);
+});
+
+//divide
+
+it("return the remainder of one number divided by another", () => {
```
Comment (kelfro) [inline basic-javascript/project-1.spec.js]: I think getRemainder is a separate function. This one should divide one number by another.

File: basic-javascript/project-1.spec.js [hunk 2]
```diff
@@ -1,3 +1,332 @@
...
+  expect(helpers.multiplyByTen(undefined)).toBe(0);
+  expect(helpers.multiplyByTen(null)).toBe(0);
+  expect(helpers.multiplyByTen(NaN)).toBe(0);
+  expect(helpers.multiplyByTen("cat")).toBe(0);
+});
+
+it("should convert a string to a number if possible", () => {
+  expect(helpers.multiplyByTen("10")).toBe(100);
+  expect(helpers.multiplyByTen("10.5")).toBe(105);
+});
```
Comment (kelfro) [inline basic-javascript/project-1.spec.js]: I like that you test each function that requires numbers to make sure it is converting numbers written as strings into plain numbers. Nice!

File: basic-javascript/project-1.spec.js [hunk 3]
```diff
@@ -1,3 +1,332 @@
...
+  expect(helpers.getRectangleArea("5vvv", 2)).toBe(10);
+  expect(helpers.getRectangleArea(5, "2sss")).toBe(10);
+});
+
+//getTriangleArea
+
+it("return the area of a triangle as a number", () => {
+  expect(helpers.getTriangleArea(5, 2)).toBe(5);
+  expect(helpers.getTriangleArea(3.55, 2.55)).toBeCloseTo(4.53);
+  expect(helpers.getTriangleArea(-2, 2)).toBe(-2);
```
Comment (kelfro) [inline basic-javascript/project-1.spec.js]: Good job using toBeCloseTo here and with getCircleArea, which involves pi. This way, you don't have to worry about how many decimal places javascript is going to calculate.

**Comment by kelfro** [comment]
Excellent work on Project-1! I love how you edited the project-1.js file to help sanitize data entry and wrote complete tests for all of the functions. I think the quality of your work here is outstanding and I know if you had had more time your project-2 would have turned out great, too.
